aten/src/THC/THCDeviceUtils.cuh

```diff
@@ -39,6 +41,15 @@ __device__ __forceinline__ T doLdg(const T* p) {
 #endif
 }
 
+template <>
+__device__ __forceinline__ c10::BFloat16 doLdg<c10::BFloat16>(const c10::BFloat16* p) {
+#if __CUDA_ARCH__ >= 350
```

Comment:
do you need #if here? torch is only supported on __CUDA_ARCH__ >= 350

Reply:
I think this is actually equivalent to `#ifndef __HIP_PLATFORM_HCC__`?

Reply:
Then it should say so? `__ldg` doesn't provide performance benefit these days, but I guess you still need to load short and construct bfloat16 from bits on cuda, and hip is able to handle it natively?

Reply:
Yes, I will change this to `#ifndef __HIP_PLATFORM_HCC__`, and maybe remove it later (needs benchmark). On HIP, it is just `*p`, so it's OK.

Reply:
fixed

Reply:
Just to let you know, changing the #if does break the build for NVIDIA GRID K520 GPU.  I understand that is not a supported CUDA architecture.

Reply:
Sorry about that, but as you note it is not a supported architecture.